src/main/java/com/weavers/duqhun/business/impl/AouthServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.weavers.duqhun.business.impl;
+
+import com.weavers.duqhun.business.AouthService;
+import com.weavers.duqhun.dao.UserAouthDao;
+import com.weavers.duqhun.dao.UsersDao;
+import com.weavers.duqhun.domain.UserAouth;
+import com.weavers.duqhun.domain.Users;
+import com.weavers.duqhun.dto.AouthBean;
+import com.weavers.duqhun.util.RandomCodeGenerator;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ *
+ * @author Android-3
+ */
+public class AouthServiceImpl implements AouthService {
+
+    @Autowired
+    UserAouthDao userAouthDao;
+    @Autowired
+    UsersDao usersDao;
+
+    private String createToken(Long userId) {
+        String token = null;
+        Calendar cal = Calendar.getInstance();
+        if (userId != null) {
+            token = RandomCodeGenerator.getNumericCode(4) + userId + cal.getTimeInMillis(); // generate new token
+        }
+        return token;
+    }
+
+    private Date nextUpdate() {
+        Calendar calendar = Calendar.getInstance(); // set token validity
+        calendar.add(Calendar.DATE, 1);
+        return calendar.getTime();
+    }
+
+    @Override
+    public AouthBean generatAccessToken(String email, Long userId) {
+        AouthBean aouthBean = new AouthBean();
+        UserAouth userAouth = userAouthDao.getTokenByMailAndUserId(email, userId);
+        String token = this.createToken(userId);
+        if (userAouth != null) {    // for existing user update token
+            userAouth.setAouthToken(token);
+            userAouth.setValidTill(nextUpdate());
+            userAouthDao.save(userAouth);
+        } else {                    // for new user create new token
+            UserAouth userAouth2 = new UserAouth();
+            userAouth2.setId(null);

```

Comment:
Id is null just because of inserting a new row in db and id is auto increment-able. If it gets null value it will increase the id and insert into the new row if id is set by a valid id then it will update respective row.